docs/port-karpenter-on-our-cloud.md

```diff
@@ -0,0 +1,90 @@
+# Porting Karpenter to your cloud provider
+
+## Pre-requisites
+
+Karpenter requires your to implement an interface to use your own cloud provider.
+The interface is defined in the `pkg/cloudprovider/types.go` file.
+
+```go
+// CloudProvider interface is implemented by cloud providers to support provisioning.
+type CloudProvider interface {
+	// Create launches a NodeClaim with the given resource requests and requirements and returns a hydrated
+	// NodeClaim back with resolved NodeClaim labels for the launched NodeClaim
+	Create(context.Context, *v1beta1.NodeClaim) (*v1beta1.NodeClaim, error)
+	// Delete removes a NodeClaim from the cloudprovider by its provider id
+	Delete(context.Context, *v1beta1.NodeClaim) error
+	// Get retrieves a NodeClaim from the cloudprovider by its provider id
+	Get(context.Context, string) (*v1beta1.NodeClaim, error)
+	// List retrieves all NodeClaims from the cloudprovider
+	List(context.Context) ([]*v1beta1.NodeClaim, error)
+	// GetInstanceTypes returns instance types supported by the cloudprovider.
+	// Availability of types or zone may vary by nodepool or over time.  Regardless of
+	// availability, the GetInstanceTypes method should always return all instance types,
+	// even those with no offerings available.
+	GetInstanceTypes(context.Context, *v1beta1.NodePool) ([]*InstanceType, error)
+	// IsDrifted returns whether a NodeClaim has drifted from the provisioning requirements
+	// it is tied to.
+	IsDrifted(context.Context, *v1beta1.NodeClaim) (DriftReason, error)
+	// Name returns the CloudProvider implementation name.
+	Name() string
+}
+```
+
+## Implementing the CloudProvider interface
+
+### Create method
+
+The `Create` method will receive a NodeClaim object and must implement the logic to
+create a compute ressource which will trigger a `Node` object creation in your cluster.
+For example, it can create a virtual machine on your cloud provider which will auto-register
+in your Kubernetes cluster.
+
+To have NodeClaim properly hydrated, you must set the `.status.providerID` and
```

Comment:
This is really great! I wonder if it makes sense to reference the Kwok cloudprovider and code samples throughout this since that's our "neutral", in-tree implementation that we can point contributors to for reference